apps/speed.c

```diff
@@ -3392,6 +3395,14 @@ int speed_main(int argc, char **argv)
                 EVP_PKEY_CTX_free(sm2_vfy_pctx);
                 break;
             }
+            if (!EVP_PKEY_sign_init(sm2_pctx)
+                || !EVP_PKEY_sign(sm2_pctx, NULL, &loopargs[i].sigsize,
+                                  NULL, 0)) {
+                EVP_PKEY_CTX_free(pctx);
+                EVP_PKEY_CTX_free(sm2_vfy_pctx);
+                break;
+            }
```

Comment:
Why are you adding this?

Reply:
I like this, 'cause it gets the *actual* signature size. `ECDSA_size` only gives a guesstimate, and relies on the assumption that a signature will never be bigger than the ECDSA one.

Reply:
If people end up doing this a lot, we might want to considering adding a public function that computes the signature size for a given key och key context. Maybe there already is one? I forget...

Reply:
An EVP_PKEY API for this is probably sensible.

Reply:
`EVP_PKEY_size()`?

Reply:
I feel a bit embarrassed at having missing this call.
I've changed now.